lisa/sut_orchestrator/azure/platform_.py

```diff
@@ -917,7 +918,9 @@ def _create_deployment_parameters(
                 )
                 arm_parameters.enable_sriov = False
 
-        template = self._load_template()
+        # the arm template may be updated by the hooks, so make a copy to avoid
+        # the original template is modified.
+        template = deepcopy(self._load_template())
```

Comment:
Can we modify the `_load_template` function to return a deepcopy instead?

Reply:
The method use a cache decorator, so the return value of this method is managed by another package. The deepcopy inside of the `_load_template` is called once like the method. So it doesn't work. The copy should happen after get the cached value.

Reply:
Got it! So why can't we remove the `lru_cache` decorator in `_load_template` function? Will it significantly impact the runtime or is it because of concurrency issues?

Reply:
It loads file from disk, I hope to reduce the disk IO, it's not a big impact although.

Reply:
If this is the case, it would be up to you to make a call. The only upside I can think of removing `lru_cache` is that we don't need to explicitly remember to `deepcopy`.